langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java

```diff
@@ -569,7 +621,18 @@ public Builder customHeaders(Map<String, String> customHeaders) {
             return this;
         }
 
+        public Builder supportedCapabilities(Set<Capability> capabilities) {
+            this.capabilities = capabilities;
+            return this;
+        }
+
         public AzureOpenAiChatModel build() {
+            if (this.capabilities == null) {
+                capabilities = new HashSet<>();
+            }
+            if (this.responseFormat != null) {
+                this.chatCompletionsResponseFormat = toAzureOpenAiResponseFormat(this.responseFormat, this.strictJsonSchema);
```

Comment:
I guess the same logic should be applied to the streaming model, it can be added in a separate PR though

Reply:
I added this as a note in https://github.com/langchain4j/langchain4j/issues/2241 so we don't forget